subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -188,7 +189,8 @@ private void configureTargetPlatform(SourceSet sourceSet, ConfigurationContainer
             compileTask.getOptions().getHeaderOutputDirectory().convention(target.getLayout().getBuildDirectory().dir(generatedHeadersDir));
             JavaPluginExtension javaPluginExtension = target.getExtensions().getByType(JavaPluginExtension.class);
             compileTask.getModularity().getInferModulePath().convention(javaPluginExtension.getModularity().getInferModulePath());
-            compileTask.getJavaCompiler().convention(getToolchainTool(target, JavaToolchain::getJavaCompiler));
+            BiFunction<JavaToolchainService, JavaToolchainSpec, Provider<JavaCompiler>> providerBiFunction = JavaToolchainService::compilerFor;
```

Comment:
I think it would be easier to read with the local variable (and the `BiFunction` type signature)

Reply:
I'll assume you meant _without_ the local variable, and indeed leaving it was a mistake.